src/model/collection/periodic-log.ts

```diff
@@ -0,0 +1,57 @@
+import assert from "assert";
+import { kebabCase } from "lodash";
+import { DateTime, DateTimeOptions, Duration, DurationLike, Interval, IntervalMaybeValid } from "luxon";
+import { parse } from "path";
+
+import { Collection } from "./schema";
+
+export class PeriodicLog extends Collection {
+    public readonly id: string;
+    public readonly folder: string;
+    public readonly dateFormat: string;
+    public readonly dateOptions: DateTimeOptions;
+    public readonly dateOffset: Duration<true>;
+    public readonly intervalDuration: Duration<true>;
+
+    public constructor(
+        id: string,
+        folder: string,
+        dateFormat: string,
+        intervalDurationLike: DurationLike,
+        dateOffsetLike: DurationLike = 0,
+        dateOptions: DateTimeOptions = {},
+    ) {
+        super();
+
+        const intervalDuration = Duration.fromDurationLike(intervalDurationLike);
+        const dateOffset = Duration.fromDurationLike(dateOffsetLike);
+        assert(id.trim().length > 0, "id must be non-empty");
+        assert(dateFormat.length > 0, "fileNameDateFormat must be non-empty");
+        assert(folder.trim().length > 0, "folder must be non-empty");
+        assert(intervalDuration.isValid && intervalDuration.valueOf() > 0, "intervalDuration must be positive");
+        assert(dateOffset.isValid && dateOffset.valueOf() >= 0, "dateOffset must be non-negative");
+
+        this.id = kebabCase(id);
+        this.folder = folder.trim();
+        this.dateFormat = dateFormat;
+        this.intervalDuration = Duration.fromDurationLike(intervalDurationLike);
+        this.dateOffset = Duration.fromDurationLike(dateOffsetLike);
```

Comment:
Use existing durations:

https://github.com/brianrodri/objo/blob/94e32c547b3a112066158de560e733f7e5c61ac0/src/model/collection/periodic-log.ts#L26-L27

```suggestion
        this.intervalDuration = intervalDuration;
        this.dateOffset = dateOffset;
```